src/tools/singlejar/token_stream_test.cc

```diff
@@ -79,6 +80,42 @@ TEST(TokenStreamTest, CommandFile) {
   EXPECT_TRUE(token_stream.AtEnd());
 }
 
+#ifdef _WIN32
+// '-foo @commandfile -bar' command line.
+TEST(TokenStreamTest, CommandFileLongPath) {
+  const char *tempdir = getenv("TEST_TMPDIR");
+  ASSERT_NE(nullptr, tempdir);
+
+  std::string command_file_path = singlejar_test_util::OutputFilePath(std::string(260, 'A'));
```

Comment:
@laszlocsomor If I changed the filename from 260 `'A'` to 250 `'A'`, the test will pass. Do you know if long path support of Win32 API has the restriction that each path cannot exceed certain length even with `\\?`' prefix? Don't see anything from https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file

Reply:
Yes, individual path components may not exceed ~260 characters. (Can't remember the exact limit and can't find where I read it :( )

Reply:
> Because you cannot use the "\\?\" prefix with a relative path, relative paths are always limited to a total of MAX_PATH characters.

Seems like this implies that each path component of an UNC-prefixed absolute path cannot exceed `MAX_PATH` (255). At least this is the maximum value I can set until the test starts to fail.

Reply:
Yes, that's my understanding.

Reply:
I suggest testing long path behavior by creating directories like "aaaaaaaa.aaa\bbbbbbbb.bbb\cccccccc.ccc\..."

These cannot be shortened to 8dot3 style so you can reliably compose a long path.

Reply:
> This type of path is composed of components separated by backslashes, each up to the value returned in the lpMaximumComponentLength parameter of the GetVolumeInformation function (this value is commonly 255 characters).

Found on https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#fully-qualified-vs-relative-paths

Reply:
> I suggest testing long path behavior by creating directories like "aaaaaaaa.aaa\bbbbbbbb.bbb\cccccccc.ccc..."

Slightly more troublesome since I need to recursively create the directories with `_wmkdir`, but also done now.